--- a/src/xvdr/xvdrclient.c
+++ b/src/xvdr/xvdrclient.c
@@ -1515,7 +1515,6 @@ bool cXVDRClient::processRECORDINGS_GetDiskSpace() /* OPCODE 100 */
 
 bool cXVDRClient::processRECORDINGS_GetCount() /* OPCODE 101 */
 {
-  Recordings.Load();
   m_resp->put_U32(Recordings.Count());
 
   return true;

--- a/src/recordings/recordingscache.c
+++ b/src/recordings/recordingscache.c
@@ -31,7 +31,6 @@
 cRecordingsCache::cRecordingsCache() : m_changed(false)
 {
   // initialize cache
-  Recordings.Load();
   for (cRecording *recording = Recordings.First(); recording; recording = Recordings.Next(recording))
     Register(recording);
 }
