--- vdr-plugin-softhddevice/Makefile	2013-03-11 16:52:57.971314857 +0100
+++ b/Makefile	2013-03-11 16:57:08.024641579 +0100
@@ -32,8 +32,8 @@
 #CONFIG += -DHAVE_PTHREAD_NAME		# supports new pthread_setname_np
 #CONFIG += -DNO_TS_AUDIO		# disable ts audio parser
 #CONFIG += -DUSE_TS_VIDEO		# build new ts video parser
-#CONFIG += -DUSE_MPEG_COMPLETE		# support only complete mpeg packets
-#CONFIG += -DUSE_VDR_SPU		# use VDR SPU decoder.
+CONFIG += -DUSE_MPEG_COMPLETE		# support only complete mpeg packets
+CONFIG += -DUSE_VDR_SPU		# use VDR SPU decoder.
 
 ifeq ($(ALSA),1)
 CONFIG += -DUSE_ALSA
